sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAdministrationUtil.java

```diff
@@ -0,0 +1,383 @@
+
+        return new TransformedResponse<>(keyVaultRoleDefinition, response);
+    }
+
+    static KeyVaultRoleDefinition roleDefinitionToKeyVaultRoleDefinition(RoleDefinition roleDefinition) {
+        List<KeyVaultPermission> keyVaultPermissions = new ArrayList<>();
+
+        for (Permission permission : roleDefinition.getPermissions()) {
+            keyVaultPermissions.add(
+                new KeyVaultPermission(permission.getActions(), permission.getNotActions(),
+                    permission.getDataActions().stream()
+                        .map(dataAction -> KeyVaultDataAction.fromString(dataAction.toString()))
+                        .collect(Collectors.toList()),
+                    permission.getNotDataActions().stream()
+                        .map(notDataAction -> KeyVaultDataAction.fromString(notDataAction.toString()))
+                        .collect(Collectors.toList())));
+        }
+
+        return new KeyVaultRoleDefinition(roleDefinition.getId(), roleDefinition.getName(),
+            KeyVaultRoleDefinitionType.fromString(roleDefinition.getType().toString()), roleDefinition.getRoleName(),
+            roleDefinition.getDescription(), KeyVaultRoleType.fromString(roleDefinition.getRoleType().toString()),
+            keyVaultPermissions, roleDefinition.getAssignableScopes().stream()
+            .map(roleScope -> KeyVaultRoleScope.fromString(roleScope.toString()))
+            .collect(Collectors.toList()));
+    }
+
+    static PagedResponse<KeyVaultRoleAssignment> transformRoleAssignmentsPagedResponse(
+        PagedResponse<RoleAssignment> pagedResponse) {
+
+        List<KeyVaultRoleAssignment> keyVaultRoleAssignments = new ArrayList<>();
+
+        for (RoleAssignment roleAssignment : pagedResponse.getValue()) {
+            keyVaultRoleAssignments.add(roleAssignmentToKeyVaultRoleAssignment(roleAssignment));
+        }
+
+        return new TransformedPagedResponse<>(keyVaultRoleAssignments, pagedResponse);
+    }
+
+    static Response<KeyVaultRoleAssignment> transformRoleAssignmentResponse(Response<RoleAssignment> response) {
+        KeyVaultRoleAssignment keyVaultRoleAssignment = roleAssignmentToKeyVaultRoleAssignment(response.getValue());
+
+        return new TransformedResponse<>(keyVaultRoleAssignment, response);
+    }
+
+    static KeyVaultRoleAssignment roleAssignmentToKeyVaultRoleAssignment(RoleAssignment roleAssignment) {
+        RoleAssignmentPropertiesWithScope propertiesWithScope = roleAssignment.getProperties();
+
+        return new KeyVaultRoleAssignment(roleAssignment.getId(), roleAssignment.getName(), roleAssignment.getType(),
+            new KeyVaultRoleAssignmentProperties(propertiesWithScope.getRoleDefinitionId(),
+                propertiesWithScope.getPrincipalId(),
+                KeyVaultRoleScope.fromString(propertiesWithScope.getScope().toString())));
+    }
+
+    private static final class TransformedPagedResponse<L extends List<T>, T, U> implements PagedResponse<T> {
```

Comment:
Similar comment for this being a duplicate of `PagedResponseBase`: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedResponseBase.java#L53

Reply:
This PR just refactors this class.
Will let @vcolin7 do logic changes to this class usage, in a follow up PR.